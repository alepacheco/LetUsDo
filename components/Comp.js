function Home() {
  return <div>Imported</div>;
}

export default Home;