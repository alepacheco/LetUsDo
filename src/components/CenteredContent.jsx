import React from 'react';
import '../styles/components/CenteredContent.scss';

export const CenteredContent = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
