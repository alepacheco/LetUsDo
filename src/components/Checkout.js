import React from 'react';
import StripeCheckout from 'react-stripe-checkout';
import Button from 'react-bootstrap/Button';

export default class Checkout extends React.Component {
    constructor(props) {
        super(props);

        this.onToken = this.onToken.bind(this);
    }

    onToken(token) {
        fetch('/save-stripe-token', {
            method: 'POST',
            body: JSON.stringify(token),
        }).then(response => {
            response.json().then(data => {
                alert(`We are in business, ${data.email}`);
            });
        });
    }

    render() {
        return (
            <StripeCheckout
                name="We Do"
                description={this.props.description || 'Complete task'}
                panelLabel="Pay"
                amount={this.props.amount || 2000}
                currency="GBP"
                token={this.onToken}
                stripeKey="pk_test_TYooMQauvdEDq54NiTphI7jx"
            >
                <Button className="get-it-done-button">Get it done</Button>
            </StripeCheckout>
        )
    }
}