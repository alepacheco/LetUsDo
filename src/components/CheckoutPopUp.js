import React from 'react';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CheckoutFormWrapper from './stripe/CheckoutFormWrapper';
import { bindActionCreators } from 'redux';

import '../styles/components/CheckoutPopUp.css';
import * as actions from '../actions/checkoutPopUpActions';

const validateEmail = email => {
  // eslint-disable-next-line no-useless-escape
  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(email).toLowerCase());
}

class CheckoutPopUp extends React.Component {
  constructor(props) {
    super(props);

    this.onChangeEmail = this.onChangeEmail.bind(this);
  }

  onChangeEmail(event) {
    const emailValid = validateEmail(event.target.value);

    this.props.actions.setCheckoutEmailValid(emailValid);
    this.props.actions.setCheckoutEmail(event.target.value);
  }

  render() {
    return (
      <Modal
        show={this.props.show}
        onHide={this.props.onHide}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header closeButton>
          <Modal.Title id="contained-modal-title-vcenter">
            Confirm order
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div className="task-description-title">
            Task description
          </div>
          <div className="task-description-content">
            {this.props.taskText || 'No description specified'}
          </div>
          <Form>
            <Form.Group controlId="formBasicEmail" className="email-form">
              <Form.Control
                type="email"
                placeholder="Enter email"
                onChange={this.onChangeEmail}
                value={this.props.email} />
            </Form.Group>
          </Form>

          <CheckoutFormWrapper />
        </Modal.Body>

      </Modal>
    );
  }
}

const mapStateToProps = (state) => ({
  taskText: state.taskModal.taskText,
  email: state.checkoutPopUp.email
});

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CheckoutPopUp);

CheckoutPopUp.propTypes = {
  onHide: PropTypes.func,
  taskText: PropTypes.string,
  email: PropTypes.string,
  show: PropTypes.bool,
  actions: PropTypes.shape({
    setCheckoutEmailValid: PropTypes.func,
    setCheckoutEmail: PropTypes.func
  }),
};