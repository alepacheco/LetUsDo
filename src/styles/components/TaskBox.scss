.task-modal {
  textarea,
  select,
  input,
  button,
  button:focus {
    outline: none;
    resize: none;
    box-shadow: none !important;
  }

  .task-modal-wrapper {
    box-shadow: 3px 3px 5px 6px rgba(0, 0, 0, 0.1);
  }

  .form-wrapper {
    background: none;
    border: none;
  }

  .center-inner {
    padding: 16px;
    width: 100%;
    background-color: #f1f1f1;
    border: none;
  }

  .left-text {
    text-align: left;
  }

  .task-modal-title {
    font-size: 26px;
    font-family: JelleeRoman, Sans-Serif;
    font-weight: normal;
    font-style: normal;
  }

  .get-it-done-button {
    margin-left: auto;
  }
  .get-it-done-button-wrapper {
    margin: 26px 0 0 0;
    text-align: right;
  }
}
