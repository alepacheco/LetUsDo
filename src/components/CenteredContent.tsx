import React from 'react';
import 'src/styles/components/CenteredContent.scss';

type CenteredContentProps = {
  reverse?: boolean;
};

export const CenteredContent: React.FC<CenteredContentProps> = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
