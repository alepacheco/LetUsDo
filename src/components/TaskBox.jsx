import React from 'react';
import Card from 'react-bootstrap/Card';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import '../styles/components/TaskBox.scss';
import * as actions from '../actions/taskModalActions';
import { CenteredContent } from './CenteredContent';

export class TaskBox extends React.Component {
  constructor(props) {
    super(props);
    this.onClickGetItDone = this.onClickGetItDone.bind(this);
    this.onChangeTaskDescription = this.onChangeTaskDescription.bind(this);
  }

  async onClickGetItDone() {
    this.props.actions.setDialog(true);
    // TODO add tracking
  }

  onChangeTaskDescription(event) {
    this.props.actions.setTaskTest(event.target.value);
  }

  render() {
    const dayOfWeek = new Date().toLocaleString('en-us', { weekday: 'long' });
    const { taskText } = this.props;

    return (
      <div className="task-modal">
        <CenteredContent>
          <Card className="center-inner">
            <Card.Body>
              <Card.Title className="task-modal-title left-text">
                What can we do for you this {dayOfWeek}?
              </Card.Title>
              <Form className="form-wrapper">
                <Form.Group>
                  <Form.Control
                    onChange={this.onChangeTaskDescription}
                    value={taskText}
                    className="task-form-input"
                    as="textarea"
                    rows="3"
                    placeholder="What task do you need done?"
                  />
                </Form.Group>
              </Form>
              <div className="get-it-done-button-wrapper">
                <Button
                  className="get-it-done-button"
                  onClick={this.onClickGetItDone}
                  disabled={taskText === ''}
                >
                  Get it done
                </Button>
              </div>
            </Card.Body>
          </Card>
        </CenteredContent>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    checkoutPopup: state.taskModal.checkoutPopupOpen,
    taskText: state.taskModal.taskText
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TaskBox);

TaskBox.propTypes = {
  taskText: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    setDialog: PropTypes.func,
    setTaskTest: PropTypes.func
  }).isRequired,
  checkoutPopup: PropTypes.bool.isRequired
};
