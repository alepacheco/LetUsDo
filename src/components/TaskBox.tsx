import React from 'react';
import Card from 'react-bootstrap/Card';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import '../styles/components/TaskBox.scss';
import * as actions from '../actions/taskModalActions';
import { CenteredContent } from './CenteredContent';
import { trackEvent } from 'src/utils/analytics';

type TaskBoxProps = {
  actions: {
    setTaskText: (text: string) => void;
    setDialog: (state: string) => void;
  };
  taskText: string;
};

export const TaskBox: React.FC<TaskBoxProps> = ({ actions, taskText }) => {
  const onClickGetItDone = () => {
    actions.setDialog('open');
    trackEvent({
      category: 'click',
      action: 'get it done button clicked',
      label: `containing task: ${taskText}`
    });
  };

  const onChangeTaskDescription = (event: any) => {
    actions.setTaskText(event.target.value);
    trackEvent({
      category: 'formChange',
      action: 'TaskBox text input change',
      label: event.target.value
    });
  };

  const dayOfWeek = new Date().toLocaleString('en-us', { weekday: 'long' });

  return (
    <div className="task-modal">
      <CenteredContent>
        <Card className="center-inner">
          <Card.Body>
            <Card.Title className="task-modal-title left-text">
              What can we do for you this {dayOfWeek}?
            </Card.Title>
            <Form className="form-wrapper">
              <Form.Group>
                <Form.Control
                  onChange={onChangeTaskDescription}
                  value={taskText}
                  className="task-form-input"
                  as="textarea"
                  rows="3"
                  placeholder="What task do you need done?"
                />
              </Form.Group>
            </Form>
            <div className="get-it-done-button-wrapper">
              <Button className="get-it-done-button" onClick={onClickGetItDone}>
                Get it done
              </Button>
            </div>
          </Card.Body>
        </Card>
      </CenteredContent>
    </div>
  );
};

function mapStateToProps(state: any) {
  return {
    taskText: state.taskModal.taskText
  };
}

function mapDispatchToProps(dispatch: any) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TaskBox);
