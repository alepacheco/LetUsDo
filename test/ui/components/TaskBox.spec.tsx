/* eslint-disable no-undef */
/* eslint no-global-assign:off */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TaskBox } from 'src/components/TaskBox';

const constantDate = new Date('2017-06-13T04:41:20');

// @ts-ignore
Date = class extends Date {
  constructor() {
    super()
    return constantDate;
  }
};

describe('<TaskBox />', () => {
  it('renders', () => {
    const wrapper = shallow(
      <TaskBox
        taskText="Sample task"
        actions={{
          setTaskText: () => {},
          setDialog: () => {}
        }}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
