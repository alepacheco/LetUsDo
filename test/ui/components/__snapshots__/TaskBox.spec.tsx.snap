// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskBox /> on form type  1`] = `
<div
  className="task-modal"
>
  <Component>
    <Card
      body={false}
      className="center-inner"
    >
      <CardBody>
        <CardTitle
          className="task-modal-title left-text"
        >
          What can we do this 
          Tuesday
          ?
        </CardTitle>
        <Form
          className="form-wrapper"
          inline={false}
        >
          <FormGroup>
            <FormControl
              as="textarea"
              className="task-form-input"
              onChange={[Function]}
              placeholder="What task do you need done?"
              rows="3"
              value="Sample task"
            />
          </FormGroup>
        </Form>
        <div
          className="get-it-done-button-wrapper"
        >
          <Button
            active={false}
            className="get-it-done-button"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            Get it done
          </Button>
        </div>
      </CardBody>
    </Card>
  </Component>
</div>
`;

exports[`<TaskBox /> onClick 1`] = `
<div
  className="task-modal"
>
  <Component>
    <Card
      body={false}
      className="center-inner"
    >
      <CardBody>
        <CardTitle
          className="task-modal-title left-text"
        >
          What can we do this 
          Tuesday
          ?
        </CardTitle>
        <Form
          className="form-wrapper"
          inline={false}
        >
          <FormGroup>
            <FormControl
              as="textarea"
              className="task-form-input"
              onChange={[Function]}
              placeholder="What task do you need done?"
              rows="3"
              value="Sample task"
            />
          </FormGroup>
        </Form>
        <div
          className="get-it-done-button-wrapper"
        >
          <Button
            active={false}
            className="get-it-done-button"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            Get it done
          </Button>
        </div>
      </CardBody>
    </Card>
  </Component>
</div>
`;

exports[`<TaskBox /> renders 1`] = `
<div
  className="task-modal"
>
  <Component>
    <Card
      body={false}
      className="center-inner"
    >
      <CardBody>
        <CardTitle
          className="task-modal-title left-text"
        >
          What can we do this 
          Tuesday
          ?
        </CardTitle>
        <Form
          className="form-wrapper"
          inline={false}
        >
          <FormGroup>
            <FormControl
              as="textarea"
              className="task-form-input"
              onChange={[Function]}
              placeholder="What task do you need done?"
              rows="3"
              value="Sample task"
            />
          </FormGroup>
        </Form>
        <div
          className="get-it-done-button-wrapper"
        >
          <Button
            active={false}
            className="get-it-done-button"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            Get it done
          </Button>
        </div>
      </CardBody>
    </Card>
  </Component>
</div>
`;
